blocks/hooks/notices/index.js

```diff
@@ -0,0 +1,86 @@
+/**
+ * External dependencies
+ */
+import uuid from 'uuid/v4';
+
+/**
+ * WordPress dependencies
+ */
+import { Component, getWrapperDisplayName } from '@wordpress/element';
+import { addFilter } from '@wordpress/hooks';
+import { NoticeList } from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import { hasBlockSupport } from '../../api';
+import './editor.scss';
+
+/**
+ * Override the default edit UI to include notices if supported.
+ *
+ * @param   {function|Component} BlockEdit Original component.
+ * @returns {Component}                    Wrapped component.
+ */
+export function withNotices( BlockEdit ) {
+	class WrappedBlockEdit extends Component {
+		/** @inheritdoc */
+		constructor() {
+			super( ...arguments );
+
+			this.addNotice = this.addNotice.bind( this );
+			this.removeNotice = this.removeNotice.bind( this );
+
+			this.state = {
+				notices: [],
+			};
+		}
+
+		/**
+		* Function passed down to blocks as a prop that adds a new notice.
+		*
+		* @param {Object} notice  Notice to add.
+		*/
+		addNotice( notice ) {
+			const noticeToAdd = notice.id ? notice : { ...notice, id: uuid() };
+			this.setState( state => ( {
+				notices: [ ...state.notices, noticeToAdd ],
+			} ) );
+		}
+
+		/**
+		* Removes a notice by id.
+		*
+		* @param {string} id  Id of the notice to remove.
+		*/
+		removeNotice( id ) {
+			this.setState( state => ( {
+				notices: state.notices.filter( notice => notice.id !== id ),
+			} ) );
+		}
+
+		/** @inheritdoc */
+		render() {
+			if ( ! hasBlockSupport( this.props.name, 'notices' ) ) {
```

Comment:
From the previous comment, can't we always add the `notices` props?

Reply:
Yes, it would be an option, the supports notices is just for the property to be opt-in, e.g: blocks just receive if they ask for it.